src/uucore/src/lib/features/fs.rs

```diff
@@ -584,10 +585,47 @@ pub fn make_path_relative_to<P1: AsRef<Path>, P2: AsRef<Path>>(path: P1, to: P2)
     components.iter().collect()
 }
 
+/// Detects symlink loops by following the symlink chain.
+///
+/// # Arguments
+///
+/// * `path` - A reference to a `Path` object representing the starting symlink.
+///
+/// # Returns
+///
+/// * An `io::Result<()>` that is `Ok(())` if no symlink loop is detected, or an error with `io::ErrorKind::InvalidInput`
+///   if a loop is detected.
+pub fn detect_symlink_loop(path: &Path) -> io::Result<()> {
```

Comment:
This function feels a bit weird because in the success case (a symlink loop is detected) it returns an error. As you use this function anyway as if it returns a bool, I would just return a bool and rename it to something like `is_symlink_loop`.